### ISSUE TITLE
AStar: Path lays through obstacles

### ISSUE THREAD

**Issue by alderven** [pythonarcade/arcade#806]
AStar path lays through obstacles:

![изображение](https://user-images.githubusercontent.com/21060932/102123657-37414280-3e58-11eb-915e-eca9d0d19f42.png)

Code to reproduce:

```
import arcade

WIDTH = 800
HEIGHT = 600


class MyGame(arcade.Window):

    def __init__(self):
        super().__init__(WIDTH, HEIGHT)

        self.player = arcade.Sprite(':resources:images/animated_characters/female_person/femalePerson_idle.png', center_x=100, center_y=100)

        self.enemy = arcade.Sprite(':resources:images/animated_characters/zombie/zombie_idle.png', center_x=WIDTH-100, center_y=HEIGHT-100)

        barriers = arcade.SpriteList()
        for x in range(-100, WIDTH+100, 100):
            sprite = arcade.Sprite(':resources:images/tiles/grassCenter.png', center_x=x, center_y=HEIGHT/2)
            barriers.append(sprite)

        self.barrier_list = arcade.AStarBarrierList(moving_sprite=self.player,
                                                    blocking_sprites=barriers,
                                                    grid_size=50,
                                                    left=-int(self.player.center_x),
                                                    right=int(WIDTH-self.player.center_x),
                                                    bottom=-int(self.player.center_y),
                                                    top=int(HEIGHT-self.player.center_y))

        self.path = None

        self.sprites = arcade.SpriteList()
        self.sprites.extend([self.player, self.enemy])
        self.sprites.extend(barriers)

    def on_draw(self):
        arcade.start_render()
        self.sprites.draw()

        if self.path:
            arcade.draw_line_strip(self.path, arcade.color.RED, 10)

    def on_update(self, delta_time):

        self.path = arcade.astar_calculate_path(start_point=self.player.position,
                                                end_point=self.enemy.position,
                                                astar_barrier_list=self.barrier_list,
                                                diagonal_movement=False)


MyGame()
arcade.run()
```

**Comment by ghost** [comment]
@alderven the grid_size=50 line should be changed to 25 or something else that is smaller

also, your barriers list doesnt seem to line up, that may be part of the problem

**Comment by pvcraven** [comment]
In this example there is not a path from point A to point B. The algorithm finds the least-cost path, and barriers have a high-cost. So if there isn't a way around, it will go through. You can expand the left/right far enough to allow it to go around and the algorithm will work.

Really, there needs to be a good way to handle a no-path by either returning a cost or an empty path or None or something.

**Comment by niklasfriberg** [comment]
This commit should solve the issue by returning None when the only path is through barriers #926

**Comment by Cleptomania** [comment]
Released with 2.6.0
